### PR TITLE
Bugfix in nlp.replace_pipe

### PR DESCRIPTION
**Pull request by svlandeg**
## Description
Current bug: if a pipeline would contain more than one component (let's say 3), and you `nlp.replace_pipe` the last one, you would get an `E959` error about "Can't insert component before index 2", because that last (old) component would already be removed so the pipeline is temporarily smaller. Fixed it by using  insertion with `after` for these edge cases.

### Types of change
bug fix

## Checklist
- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/language.py [hunk 1]
```diff
@@ -771,8 +771,10 @@ def replace_pipe(
         self.remove_pipe(name)
         if not len(self.pipeline):  # we have no components to insert before/after
             self.add_pipe(factory_name, name=name)
```
Comment (ines) [inline spacy/language.py:773]: Since `after=pipe_index - 1` is equivalent to `last=True` (default behaviour), maybe we can combine those two conditions above? Then we can save that extra conditional.

**Comment by svlandeg** [comment]
@ines : I hope it's OK I went ahead and merged after addressing your comments, because I want to get another PR out that needs this fix :-)

**Comment by ines** [comment]
@svlandeg Yeah, no problem, it's a very straightforward fix so feel free to merge stuff like that yourself 😃
